packages/mui-system/src/Unstable_Grid/GridProps.ts

```diff
@@ -141,6 +141,28 @@ export interface GridBaseProps extends Breakpoints {
    * If `true`, the negative margin and padding are apply only to the top and left sides of the grid.
    */
   disableEqualOverflow?: boolean;
+  /**
+   * The level of the grid starts from `0`
+   * and increases when the grid nests inside another grid regardless of container or item.
+   *
+   * ```js
+   * <Grid> // level 0
+   *   <Grid> // level 1
+   *     <Grid> // level 2
+   *   <Grid> // level 1
+   * ```
+   *
+   * Only consecutive grid is considered nesting.
+   * A grid container will start at `0` if there are non-Grid element above it.
+   *
+   * ```js
+   * <Grid> // level 0
+   *   <div>
+   *     <Grid> // level 0
+   *       <Grid> // level 1
+   * ```
+   */
+  level?: number;
```

Comment:
What's the motivation of making this a public API? If we don't want it to be public, maybe we should prefix it with `internal_`.

Reply:
Ping on this question, it's still not clear to me.

Reply:
Okay, I will make it `unstable_` for consistency.